editor/editor_inspector.cpp

```diff
@@ -3650,6 +3673,59 @@ Variant EditorInspector::get_property_clipboard() const {
 	return property_clipboard;
 }
 
+void EditorInspector::_add_meta_confirm_callback() {
+	String name = add_meta_name->get_text();
+	if (!name.is_valid_identifier()) {
+		EditorNode::get_singleton()->show_warning(vformat(TTR("Invalid metadata identifier: '%s'"), name));
+		return;
+	}
+	if (object->has_meta(name)) {
+		EditorNode::get_singleton()->show_warning(vformat(TTR("Metadata already exists: '%s'"), name));
+		return;
+	}
+
+	object->editor_set_section_unfold("metadata", true); // Ensure metadata is unfolded when adding a new metadata.
+
+	Variant defval;
+	Callable::CallError ce;
+	Variant::construct(Variant::Type(add_meta_type->get_selected() + 1), defval, nullptr, 0, ce);
+	undo_redo->create_action("Add metadata");
+	undo_redo->add_do_method(object, "set_meta", name, defval);
+	undo_redo->add_undo_method(object, "set_meta", name, Variant());
+	undo_redo->commit_action();
+
+	add_meta_dialog->hide();
+}
+
+void EditorInspector::_add_meta_callback() {
```

Comment:
Probably not important, but this method name confused me at first. It sounds like it's adding a meta callback.